plugins/generic/pln/templates/settings.tpl

```diff
@@ -27,7 +27,7 @@
 					{if $hasIssn}
 						{foreach name=terms from=$terms_of_use key=term_name item=term_data}
 							<p>{$term_data.term}</p>
-							<input type="checkbox" name="terms_agreed[{$term_name|escape}]" value="1"{if $terms_of_use_agreement[$term_name] == 1} checked{/if}><label class="agree" for="terms_agreed[{$term_name|escape}]">{translate key="plugins.generic.pln.settings.terms_of_use_agree"}</label>
+							<input type="checkbox" name="terms_agreed[{$term_name|escape}]" value="1"{if $terms_of_use_agreement[$term_name] != null} checked{/if}><label class="agree" for="terms_agreed[{$term_name|escape}]">{translate key="plugins.generic.pln.settings.terms_of_use_agree"}</label>
```

Comment:
Generally I've found PHP's type juggling behaves most predictably and stays readable if you stick with...

    if (!$thing) ...

...or...

    if ($thing) ...

...using also isset if necessary to avoid unset warnings, e.g...

    if (isset($array['thing']) && $array['thing']) ...

Unless you really have distinctions between null/false/0/"" then coding for one of them is overly specific. If you do need them, use strict comparisons with triple-equals.